src/main/java/seedu/rc4hdb/model/resident/predicates/AttributesMatchAnyKeywordPredicate.java

```diff
@@ -27,29 +24,26 @@ public AttributesMatchAnyKeywordPredicate(ResidentStringDescriptor keywords) {
     @Override
     public boolean test(Resident resident) {
         assert resident != null : "Resident object is null";
-        return descriptor.getName().map(name -> resident.getName().contains(name)).orElse(false)
-                || descriptor.getPhone().map(phone -> resident.getPhone().contains(phone)).orElse(false)
-                || descriptor.getEmail().map(email -> resident.getEmail().contains(email)).orElse(false)
-                || descriptor.getRoom().map(room -> resident.getRoom().contains(room)).orElse(false)
-                || descriptor.getGender().map(gender -> resident.getGender().contains(gender)).orElse(false)
-                || descriptor.getHouse().map(house -> resident.getHouse().contains(house)).orElse(false)
-                || descriptor.getMatricNumber().map(matric -> resident.getMatricNumber().contains(matric))
+        return descriptor.getName().map(name -> resident.getName().containsIgnoreCase(name)).orElse(false)
```

Comment:
I think we may need containsIgnoreCase to have a string argument since the new descriptor class uses string. However, I'll check if this implementation works once the merge is done.